tmt/base.py

```diff
@@ -528,11 +524,40 @@ def environment(self):
         """ Return combined environment from plan data and command line """
         if self.my_run and self.my_run.environment:
             combined = self._environment.copy()
+
+            # vars sets in cli takes precedence
             combined.update(self.my_run.environment)
             return combined
         else:
             return self._environment
 
+    @staticmethod
+    def _get_environment_vars(node):
+        # Environment variables from files
+        env_files = node.get("environment-file") or []
+        if not isinstance(env_files, list):
+            raise tmt.utils.GeneralError(
+                f"environment-file parameter should be a list. "
+                f"Received {type(env_files)}"
+                )
+        env_files_vars = tmt.utils.environment_file_to_dict(env_files)
+        environment = env_files_vars
```

Comment:
Assign `environment` directly on the line above?

Reply:
fixed